### PR TITLE
Fix the issues of Makefile & Test Cases, add Github Actions

### PR DESCRIPTION
**Pull request by haoel**
- Fix the issue of Makefile & Test Case
- Add the Github action for code checking and run the unit test.

### CODE REVIEW

**Review by zhao-kun — APPROVED** [review]

File: operator/Makefile [hunk 1]
```diff
@@ -33,25 +40,32 @@ IMG ?= controller:latest
...
-endif
+K8S_VER=1.19.x
+
+TARGET = ${MKFILE_DIR}bin/manager
+ALL_SOURCE_FILES = $(shell find ${MKFILE_DIR} -type f -name "*.go")
 
-all: manager
 
+all: manager
+#export KUBEBUILDER_ASSETS=/Users/chenhao/GitHub/easemesh/operator/testbin/bin;
```
Comment (xxx7xxxx) [inline operator/Makefile:50]: redundant line

File: operator/Makefile [hunk 2]
```diff
@@ -33,25 +40,32 @@ IMG ?= controller:latest
...
 # Run tests
-ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
-test: generate fmt vet manifests
+ENVTEST_ASSETS_DIR=${MKFILE_DIR}/testbin
+test1: generate fmt vet manifests setup_envtest
```
Comment (xxx7xxxx) [inline operator/Makefile:53]: Propose a formal target name, such as `test_k8s_version`

File: operator/pkg/controllers/suite_test.go [hunk 3]
```diff
@@ -54,9 +55,10 @@ var _ = BeforeSuite(func() {
 	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
 
 	By("bootstrapping test environment")
+
 	testEnv = &envtest.Environment{
 		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "config", "crd", "bases")},
-		BinaryAssetsDirectory: filepath.Join("..", "..", "testbin", "bin"),
+		BinaryAssetsDirectory: os.Getenv("KUBEBUILDER_ASSETS"),
```
Comment (xxx7xxxx) [inline operator/pkg/controllers/suite_test.go:61]: Which is the place to set this env? Do we need to set it by using `export KUBEBUILDER_ASSETS=...` manually every time?

Comment (haoel) [reply operator/pkg/controllers/suite_test.go:61]: Originally, it is set by `setup-envtest.sh` script, and the new tool `setup-envtest` also set that by this command in makefile (this change is here: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.9.0)

```
source <(setup-envtest use -i -p env 1.21.x)
```

for more information for new tool, please take a look https://github.com/kubernetes-sigs/controller-runtime/tree/master/tools/setup-envtest

Comment (xxx7xxxx) [reply operator/pkg/controllers/suite_test.go:61]: Understood, thx

File: operator/Makefile (hunk 2 above)
Comment (haoel) [reply operator/Makefile:53]: this just uses the new tool. how about change to `newtest`?

Comment (xxx7xxxx) [reply operator/Makefile:53]: Good to me

**Review by xxx7xxxx — APPROVED** [review]

**Pull request merged**